File: architecturecomponents/src/main/java/com/fesskiev/architecturecomponents/domain/source/remote/retrofit/APIService.java
package com.fesskiev.architecturecomponents.domain.source.remote.retrofit;


import com.fesskiev.architecturecomponents.domain.entity.WeatherResponse;

import java.util.Map;

import io.reactivex.Observable;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface APIService {

    @GET(".")
    Single<WeatherResponse> getWeatherByCity(@QueryMap Map<String, String> options);

    @GET(".")
    Observable<WeatherResponse> getWeatherByLocation(@QueryMap Map<String, String> options);

}

File: kotlinsamples/src/main/java/com/fesskiev/kotlinsamples/domain/source/remote/ApiService.kt
package com.fesskiev.kotlinsamples.domain.source.remote

import com.fesskiev.kotlinsamples.domain.entity.TopTracks
import kotlinx.coroutines.experimental.Deferred
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface ApiService {

    @GET(".")
    fun getTopTracks(@QueryMap options: Map<String, String>): Deferred<TopTracks>
}